esignet-service/src/main/java/io/mosip/esignet/advice/HeaderValidationFilter.java

```diff
@@ -101,4 +117,38 @@ private String getMessage(String errorCode) {
         }
         return errorCode;
     }
+
+    private void validateApiRateLimits(String path, String transactionId, String individualIdHash) {
+        int apiCode = path.endsWith("send-otp") ? 1 : path.endsWith("authenticate")? 2 : 3;
+
+        switch (apiCode) {
+            case 1:
+                ApiRateLimit sendOtpLimit = cacheUtilService.getApiRateLimitTransaction(transactionId);
+                checkCountLimit(1, sendOtpLimit, sendOtpAttempts, individualIdHash);
+                cacheUtilService.saveApiRateLimit(transactionId, sendOtpLimit);
+                break;
+            case 2:
+                ApiRateLimit authenticateLimit = cacheUtilService.getApiRateLimitTransaction(transactionId);
+                checkCountLimit(2, authenticateLimit, authenticateAttempts, individualIdHash);
+                cacheUtilService.saveApiRateLimit(transactionId, authenticateLimit);
+                break;
+        }
+    }
```

Comment:
Since we are having all these conditional checks, I think we should have unit test file for this class .

Reply:
Added